apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerConnectionTests.AbortedTransactionRetryTests.cs

```diff
@@ -0,0 +1,337 @@
+            private const int ThirdCallHighBo = 2 * (250 + 500);
+
+            private const int ExceptionRetryDelayMs = 300;
+            private const int SecondCallLowEx = 300;
+            private const int SecondCallHighEx = 2 * 300;
+            private const int ThirdCallLowEx = 300 + 300;
+            private const int ThirdCallHighEx = 2 * (300 + 300);
+
+            [Fact]
+            public async Task FirstCallSucceeds()
+            {
+                var spannerClientMock = SpannerClientHelpers
+                    .CreateMockClient(Logger.DefaultLogger, MockBehavior.Strict)
+                    .SetupBatchCreateSessionsAsync()
+                    .SetupBeginTransactionAsync()
+                    .SetupExecuteBatchDmlAsync()
+                    .SetupCommitAsync();
+                SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+
+                var scheduler = (FakeScheduler)connection.Builder.SessionPoolManager.SpannerSettings.Scheduler;
+                var time0 = scheduler.Clock.GetCurrentDateTimeUtc();
+                var callee = new Callee(scheduler);
+
+                await scheduler.RunAsync(async () =>
+                {
+                    var result = await connection.RunWithRetriableTransactionAsync(transaction => callee.DatabaseWorkAsync(connection, transaction));
+                    Assert.Equal(1, result);
+                });
+
+                callee.AssertCallTimesInrange(new Tuple<DateTime, DateTime>(time0, time0));
+                Assert.Equal(time0, scheduler.Clock.GetCurrentDateTimeUtc());
+            }
+
+            [Fact]
+            public async Task CommitAbortsTwice()
+            {
+                var spannerClientMock = SpannerClientHelpers
+                    .CreateMockClient(Logger.DefaultLogger, MockBehavior.Strict)
+                    .SetupBatchCreateSessionsAsync()
+                    .SetupBeginTransactionAsync()
+                    .SetupExecuteBatchDmlAsync()
+                    .SetupCommitAsync_Fails(failures: 2, statusCode: StatusCode.Aborted)
+                    .SetupRollbackAsync();
+
+                SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+
+                var scheduler = (FakeScheduler)connection.Builder.SessionPoolManager.SpannerSettings.Scheduler;
+                var time0 = scheduler.Clock.GetCurrentDateTimeUtc();
+                var callee = new Callee(scheduler);
+
+                await scheduler.RunAsync(async () =>
+                {
+                    var result = await connection.RunWithRetriableTransactionAsync(transaction => callee.DatabaseWorkAsync(connection, transaction));
+                    Assert.Equal(3, result);
+                });
+
+                callee.AssertCallTimesInrange(
+                    new Tuple<DateTime, DateTime>(time0, time0),
+                    new Tuple<DateTime, DateTime>(time0.AddMilliseconds(SecondCallLowBo), time0.AddMilliseconds(SecondCallHighBo)),
```

Comment:
If you created TimeSpan equivalents of the constants, you could just use `time0 + SecondCallLowBackoff` etc. Get rid of 18 calls to AddMilliseconds :)